addons/website/models/theme_models.py

```diff
@@ -279,7 +279,7 @@ def write(self, vals):
         for record in self:
             # Do not mark the view as user updated if original view arch is similar
             arch = vals.get('arch', vals.get('arch_base'))
-            if record.theme_template_id and record.theme_template_id.arch == arch:
+            if record.theme_template_id and record.theme_template_id.sudo().arch == arch:
```

Comment:
I think this is fixing the consequence and not the root cause.
https://github.com/odoo/odoo/commit/1eb7c577ecb31c2f0876b6dd76769e2b47802cb0 is probably missing a `if` condition on top of that whole code to check if we are in a module update state, that's the only flow where we want to perform that check.

Reply:
@rdeodoo Thank you. It means that when i upgrade the theme, it will execute this code right? I still haven't found a sign yet to know if it's in an upgrading state the module or not while in View. Or maybe I have to add some context somewhere else? Need your more explanation.

Reply:
Sorry for the late reply, you can check with `if self.pool._init:`
Around this https://github.com/odoo/odoo/commit/1eb7c577ecb31c2f0876b6dd76769e2b47802cb0#diff-4e6ab140edc4e1362791f5184ab01d23b5a27969011a27f998c9ef804dd64d53R225
That should fix the root cause

Reply:
Done. That's so great! It was a new knowledge for me. Thank you very much! :smile:

Reply:
You are welcome